.circleci/deploy-macos.sh

```diff
@@ -33,18 +33,25 @@ export MACOSX_DEPLOYMENT_TARGET=10.12
 # tries to delete them again and causes errors. To avoid this issue, we update
 # manually here for the current main version of Python (required by pyenv).
 brew upgrade python@3.8
```

Comment:
Try this if Mac builds fail.
```suggestion
brew upgrade python@3.9
```